larpandoracontent/LArObjects/LArAdaBoostDecisionTree.cc

```diff
@@ -0,0 +1,299 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArAdaBoostDecisionTree.cc
+ *
+ *  @brief  Implementation of the lar adaptive boost decision tree class.
+ *
+ *  $Log: $
+ */
+
+#include "Helpers/XmlHelper.h"
+
+#include "larpandoracontent/LArObjects/LArAdaBoostDecisionTree.h"
+
+namespace lar_content
+{
+
+AdaBoostDecisionTree::AdaBoostDecisionTree() :
+    m_isInitialized(false)
+{
+    std::cout << "AdaBoostDecisionTree::AdaBoostDecisionTree" << std::endl;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+pandora::StatusCode AdaBoostDecisionTree::Initialize(const std::string &parameterLocation, const std::string &bdtName)
+{
+    if (m_isInitialized)
+    {
+        std::cout << "AdaBoostDecisionTree: bdt was already initialized" << std::endl;
+        return pandora::STATUS_CODE_FAILURE;
+    }
+
+    // ATTN: Add consistency check for number of features, bearing in min not all features in a bdt may be used
+    this->ReadXmlFile(parameterLocation, bdtName);
+ 
+    m_isInitialized = true;
+    return pandora::STATUS_CODE_SUCCESS;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool AdaBoostDecisionTree::Classify(const DoubleVector &features) const
+{
+    return (m_pStrongClassifier->Predict(features) > 0.0 ? true : false);
```

Comment:
What if `m_pStrongClassifier` is `not yet initialised? And in all other public functions, below.

Reply:
Check added.